sponsorship_compassion/model/contracts.py

```diff
@@ -61,8 +67,43 @@ def _get_contract_from_invoice(invoice_obj, cr, uid, invoice_ids,
                         contract.write({
                             'activation_date': datetime.today().strftime(DF)})
 
+                        # Cancel the invoices if a contract is activated
+                        self._cancel_invoices(
+                            cr, uid,
+                            invoice.partner_id.id,
+                            first_pay_date,
+                            context)
         return list(res)
 
+    def _cancel_invoices(
+            self, cr, uid, partner_id, date_invoice, context=None):

```

Comment:
You should also pass as parameter the contract_id that is activated, because you wan't to filter the invoice lines only on those for which invoice_line.contract_id.id = contract_id (i.e. you want only to cancel invoices that are for the same contract, not all invoices of the partner).